src/Mod/Sketcher/App/SketchObject.cpp

```diff
@@ -1381,68 +1382,168 @@ int SketchObject::movePoint(int GeoId, PointPos PosId, const Base::Vector3d& toP
     return lastSolverStatus;
 }
 
+template <>
+Base::Vector3d SketchObject::getPointForGeometry<>(const Part::GeomPoint *geomPoint, PointPos PosId)
+{
+    if (PosId == PointPos::start || PosId == PointPos::mid || PosId == PointPos::end)
+        return geomPoint->getPoint();
+    return Base::Vector3d();
+}
+
+template <>
+Base::Vector3d SketchObject::getPointForGeometry<>(const Part::GeomLineSegment *lineSeg, PointPos PosId)
+{
+    switch (PosId) {
+    case PointPos::start: {
+        return lineSeg->getStartPoint();
+    }
+    case PointPos::end: {
+        return lineSeg->getEndPoint();
+    }
+    default:
+        break;
+    }
+    return Base::Vector3d();

```

Comment:
return could be inside the default

Reply:
I'd like to leave it outside just as a matter of preference to have some return outside.